core/src/main/java/io/kestra/core/tasks/flows/ForEachItem.java

```diff
@@ -0,0 +1,308 @@
+import io.kestra.core.models.tasks.VoidOutput;
+import io.kestra.core.runners.ExecutableUtils;
+import io.kestra.core.runners.FlowExecutorInterface;
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.runners.RunnerUtils;
+import io.kestra.core.runners.WorkerTaskExecution;
+import io.kestra.core.runners.WorkerTaskResult;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import lombok.experimental.SuperBuilder;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Positive;
+
+@SuperBuilder(toBuilder = true)
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "Execute a subflow for each batch of items",
+    description = "This tasks allow to execute a subflow for each batch of items. The items must come from Kestra's internal storage."
+)
+@Plugin(
+    examples = {
+        @Example(
+            title = "Execute a subflow for each batch of items",
+            code = {
+                """
+                    id: each
+                    type: io.kestra.core.tasks.flows.ForEachItem
+                    items: "{{ outputs.extract.uri }}" # works with API payloads too. Kestra can detect if this output is not a file,\s
+                    # and will make it to a file, split into (batches of) items
+                    maxItemsPerBatch: 10
+                    subflow:
+                      flowId: file
+                      namespace: dev
+                      inputs:
+                        file: "{{ taskrun.items }}" # special variable that contains the items of the batch
+                      wait: true # wait for the subflow execution
+                      transmitFailed: true # fail the task run if the subflow fail"""
+            }
+        )
+    }
+)
+public class ForEachItem extends Task implements ExecutableTask<VoidOutput> {
+    private static final String URI_FORMAT = "kestra:///%s/%s/executions/%s/tasks/%s/%s/bach-%s.ion";
+
+    @NotEmpty
+    @PluginProperty(dynamic = true)
+    @Schema(title = "The items, must be an URI from Kestra's internal storage")
+    private String items;
+
+    @Positive
+    @NotNull
+    @PluginProperty
+    @Builder.Default
+    @Schema(title = "Maximum number of items per batch")
+    private Integer maxItemsPerBatch = 10;
+
+    @NotNull
+    @PluginProperty
+    @Schema(title = "The subflow that will be executed on each batch of items")
+    private SubFlow subFlow;
```

Comment:
```suggestion
    private subflow subflow;
```

can we name it subflow or subflowId?

Reply:
Done